src/extensions/users-permissions/server/services/providers-registry.js

```diff
@@ -87,14 +87,21 @@ const getInitialProviders = ({ purest }) => ({
         email: body.email,
       }));
   },
-  async google({ accessToken }) {
+  async google({ accessToken, ctx }) {
     const google = purest({ provider: 'google' });
 
     const { body } = await google
       .get("https://www.googleapis.com/oauth2/v3/userinfo")
       .auth(accessToken)
       .request();
 
+    const authenticatedUserId = ctx.state.user?.id;
+
+    // TODO: Validation check if the profile is already created
+    // hasProfileAssociated ?
```

Comment:
let's query for a user with this email to determine if the user has already been created... we can get the email from `body.email`